crates/primitives/starknet/src/transaction/mod.rs

```diff
@@ -327,58 +329,82 @@ impl Transaction {
     ///
     /// * `TransactionExecutionResult<TransactionExecutionInfo>` - The result of the transaction
     ///   execution
-    pub fn execute<S: State>(
+    pub fn execute<S: State + StateChanges>(
         &self,
         state: &mut S,
         block: StarknetBlock,
         tx_type: TxType,
         contract_class: Option<ContractClass>,
         fee_token_address: ContractAddressWrapper,
-    ) -> TransactionExecutionResultWrapper<Option<CallInfo>> {
+    ) -> TransactionExecutionResultWrapper<TransactionExecutionInfoWrapper> {
         // Create the block context.
-        let block_context = BlockContext::try_serialize(block.header().clone(), fee_token_address)
+        // TODO: don't do that.
+        let mut block_context = BlockContext::try_serialize(block.header().clone(), fee_token_address)
             .map_err(|_| TransactionExecutionErrorWrapper::BlockContextSerializationError)?;
+        // TODO: use real value.
+        block_context.gas_price = 10;
         // Initialize the execution resources.
         let execution_resources = &mut ExecutionResources::default();
 
         // Verify the transaction version.
         self.verify_tx_version(&tx_type)?;
-
-        match tx_type {
-            TxType::InvokeTx => {
+        // TODO: Investigate the use of tx.execute() instead of tx.run_execute()
```

Comment:
I didn't add that idk if an issue exists for it yet. @EvolveArt can you enlighten me ?

Reply:
We can remove this todo now it's pretty clear that it was the right approach